python/ray/scripts/scripts.py

```diff
@@ -42,6 +42,21 @@ def check_no_existing_redis_clients(node_ip_address, redis_client):
                             "clients with this IP address.")
 
 
+def include_java_from_redis(redis_client):
+    """This is used for query include_java bool from redis.
+    """
+    include_java_value = redis_client.get("INCLUDE_JAVA")
```

Comment:
Actually, I don't see why we need to store anything in Redis. Can't we get rid of this key?

The flag is already part of `ray_params`. And we need to pass `--include-java` into `ray start` on every machine, but that seems like the right way to do it.

Reply:
Passing the flag `--include-java` to `ray start` command on every machine is dangerous!
We must make sure that the value of this flag on each machine is the same.

If `Node1`'s `--include-java` is true, and `Node2`s' is false, when user run a Java driver on `Node1`, the Java task may forward to `Node2`, and `Node2` will fail to start a Java worker because of the incorrect java worker command.

I think the best practice for this is to specify this flag for head node, and all worker nodes follow the head.